webpack/ForemanTasks/Components/TasksTable/TasksTableReducer.js

```diff
@@ -0,0 +1,53 @@
+import Immutable from 'seamless-immutable';
+import { combineReducers } from 'redux';
+import { createTableReducer } from 'foremanReact/components/common/table';
+import {
+  TASKS_CONTROLLER,
+  TASKS_SUCCESS,
+  SET_SORT_TASKS_TABLE,
+  SET_PAGINATION_TASKS_TABLE,
+} from './TasksTableConstants';
+
+const initialState = Immutable({
+  query: null,
+  itemCount: 0,
+  pagination: {
+    page: 1,
+    perPage: 20,
+  },
+});
+
+export const TasksTableQueryReducer = (state = initialState, action) => {
+  const { type, payload } = action;
```

Comment:
can you destruct `payload` into `page, per_page, by, order`?

Reply:
I will have to split it to `{subtotal, page, per_page, by, order, perPage } ` and except for `page` none are reused, are you sure it's a good idea? :)

Reply:
@MariaAga why not just `const { page, per_page, by, order } = payload;` ? You don't have to extract every value if you don't need them..

Reply:
I meant that each one of the values `{subtotal, page, per_page, by, order, perPage } ` is used once in the reducer (in different cases), so I'm not sure that destructing payload will be more readable. Also for each case only 1 or 2 values are used so I think its less readable to destruct all the 6 values

Reply:
This is only cosmetic, I prefer not calling the `payload` object each time,
it is not a blocker..

Reply:
I tried it but it causes errors as payload can be undefined and you can't put `const {...} = ...` inside switch case

Reply:
Done, thanks!